frontend/views/containers/chatroom/ChatMembersAllModal.vue

```diff
@@ -270,6 +312,51 @@ export default ({
       }
     },
     async addToChannel (username: string, undoing = false) {
+      if (this.isDirectMessage()) {
+        if (this.isPrivacyLevelPrivate) {
+          const usernames = [this.usernameFromDirectMessageID(this.currentChatRoomId), username]
+          const existingChatRoomId = Object.keys(this.ourGroupDirectMessages).find(contractID => {
+            const users = Object.keys(this.$store.state[contractID]?.users || {})
+            if (users.length === usernames.length + 1) {
+              return [...usernames, this.ourUsername].reduce((existing, un) => existing && users.includes(un), true)
+            }
+            return false
+          })
+          if (existingChatRoomId) {
+            this.$router.push({ name: 'GroupChatConversation', params: { chatRoomId: existingChatRoomId } })
+          } else {
+            sbp('gi.actions/mailbox/createDirectMessage', {
+              contractID: this.currentIdentityState.attributes.mailbox,
+              data: {
+                privacyLevel: CHATROOM_PRIVACY_LEVEL.GROUP,
+                usernames
+              }
+            })
+          }
+          this.closeModal()
+        } else {
+          const profile = this.ourContactProfiles[username]
+          await sbp('gi.actions/chatroom/join', {
+            contractID: this.currentChatRoomId,
+            data: { username },
+            hooks: {
+              postpublish: (message) => {
+                sbp('gi.actions/mailbox/joinDirectMessage', {
+                  contractID: profile.mailbox,
+                  data: {
+                    privacyLevel: CHATROOM_PRIVACY_LEVEL.GROUP,
+                    contractID: this.currentChatRoomId
+                  }
+                })
+              }
```

Comment:
This seems like it could be simply placed after the first call to `await sbp('gi.actions/chatroom/join'` instead of being inside of `postpublish`

Reply:
Okay.